unit_test.py

```diff
@@ -411,7 +459,121 @@ def test_services_table(self):
         # Assert
         self.assertDictEqual(result, expected)
 
-        
-
+    def test_bgp_as_detection(self):

```

Comment:
I think it’s also best to make sure each test case has a clear and specific purpose. Separating out each CNI detection into their own tests is more explicit and understandable.